### PR TITLE
Add etcd to production eks

### LINKED ISSUE #10: Add etcd to addons
## ETCD 

Follow existing pattern and add another separate **etcd repo** under **addon/** that has install and uninstall etcd-operator to Kubernetes via a helm package.

### PR DESCRIPTION
**Pull request by nehayward**
# ETCD for Production EKS

This install etcd using helm, I looked into using [etcd-operator](https://github.com/kubernetes/charts/tree/master/stable/etcd-operator) but it's still in beta so decided against it for now.

## Testing

Run terraform apply, you can then either expose the etcd service or you can port-forward service.
1. `kubectl port-forward svc/etcd  9090:2379`
2. `curl localhost:9090/v2/keys/message -X PUT -d value="Hello world"`
3. `curl localhost:9090/v2/keys/message" to verify

closes #10

### CODE REVIEW

File: production_eks_cluster/sso.tf [hunk 1]
```diff
@@ -1,7 +1,4 @@
 resource "null_resource" "sso" {
-  # add sso to cluster
-  name = "${var.cluster_name}"
```
Comment (nehayward) [inline production_eks_cluster/sso.tf]: Conflicted with main.tf variable.

**Comment by zpatrick** [comment]
This will need a section in the readme describing how to use/get to the etcd cluster.

**Comment by zpatrick** [comment]
@diemonster @nehayward I'm not sure why this is being added to this example? Seems like something I'd expect to see in the addons repo. Unless someone had built their own service that requires etcd, why would someone want this in their cluster by default?

**Comment by nehayward** [comment]
Im not sure what the typical use case would be. What should be explained? Besides updating that it’s accesible.  @bivers

**Comment by diemonster** [comment]
@zpatrick we discussed where to put this in the Monday meeting, and I thought we agreed that the `addons` repo was basically redundant? 

Why would someone want `etcd` in their default production EKS cluster? I don't know, but we want it in _our_ production EKS cluster.

**Comment by diemonster** [comment]
@nehayward can you explain the difference between `etcd-operator` and this distribution? What are the pros/cons of using either? Beta doesn't scare me, tbh.

My initial understanding is that `etcd-operator` includes a ton of automated administration tasks, such as backups of the k/v store, failover, rolling upgrades, etc. These are things I think we need, especially as we don't have the resources to administer `etcd` in an old-fashioned way.

Also agree that there should be some sort of README section that describes simple use cases of inserting and reading k/v pairs and seeing how versioning works, etc. Even if it just directs the user to an outside introduction doc.

**Pull request closed without merging**